staging/src/k8s.io/component-base/logs/api/v1/kube_features.go

```diff
@@ -57,16 +58,22 @@ const (
 	LoggingStableOptions featuregate.Feature = "LoggingStableOptions"
 )
 
-func featureGates() map[featuregate.Feature]featuregate.FeatureSpec {
-	return map[featuregate.Feature]featuregate.FeatureSpec{
-		ContextualLogging: {Default: contextualLoggingDefault, PreRelease: featuregate.Beta},
-
-		LoggingAlphaOptions: {Default: false, PreRelease: featuregate.Alpha},
-		LoggingBetaOptions:  {Default: true, PreRelease: featuregate.Beta},
+func featureGates() map[featuregate.Feature]featuregate.VersionedSpecs {
+	return map[featuregate.Feature]featuregate.VersionedSpecs{
+		ContextualLogging: {
+			{Version: version.MustParse("1.24"), Default: contextualLoggingDefault, PreRelease: featuregate.Alpha},
+			{Version: version.MustParse("1.30"), Default: contextualLoggingDefault, PreRelease: featuregate.Beta},
+		},
+		LoggingAlphaOptions: {
+			{Version: version.MustParse("1.24"), Default: false, PreRelease: featuregate.Alpha},
+		},
+		LoggingBetaOptions: {
+			{Version: version.MustParse("1.24"), Default: true, PreRelease: featuregate.Beta},
+		},
```

Comment:
moving ContextualLogging to versioned is good, but isn't it more confusing to convert LoggingAlphaOptions / LoggingBetaOptions to versioned, since they will eternally be alpha and beta and will never change?